modules/programs/git.nix

```
@@ -415,12 +428,28 @@ in {
       (filterAttrs hasSmtp config.accounts.email.accounts);
     }
 
-    (mkIf (cfg.signing != null) {
-      programs.git.iniContent = {
-        user.signingKey = mkIf (cfg.signing.key != null) cfg.signing.key;
-        commit.gpgSign = mkDefault cfg.signing.signByDefault;
-        tag.gpgSign = mkDefault cfg.signing.signByDefault;
-        gpg.program = cfg.signing.gpgPath;
+    (mkIf (cfg.signing != { }) {
+      programs.git = {
+        signing = {
+          format = mkIf (versionOlder config.home.stateVersion "24.11")
+            (mkOptionDefault "openpgp");
+          signer = mkIf (cfg.signing.format != null) (mkOptionDefault {
+            openpgp = getExe' pkgs.gnupg "gpg2";
+            ssh = getExe pkgs.ssh;
```

I think the `ssh` package is`pkgs.openssh`, and according the documentation it should be `ssh-keygen`:
>The default value for gpg.x509.program is "gpgsm" and gpg.ssh.program is "ssh-keygen". [Source](https://git-scm.com/docs/git-config#Documentation/git-config.txt-gpgltformatgtprogram)
```suggestion
            ssh = getExe' pkgs.openssh "ssh-keygen";
```